server/src/models/internshipModule.ts

```diff
@@ -173,6 +141,11 @@ InternshipModuleSchema.methods.acceptPostponement = async function (
   if (reason) event.comment = reason;
   this.events.push(event);
   this.status = InternshipModuleStatuses.PLANNED;
+  const recentPostponementRequest = this.events
+    .filter((e) => e.changes?.newSemester)
+    .reduce((event, next) => ((event.timestamp ?? 0) > (next.timestamp ?? 0) ? event : next));
+  this.inSemester = recentPostponementRequest.changes?.newSemester as string;
```

Comment:
ach das meintest du vorhin. also wird hier erst die property inSemester und inSemesterOfStudy gesetzt, beim accepten nicht schon beim requesten. ok ja, dann sollte in das event noch die changes rein!